layerindex/tools/import_otherdistro.py

```diff
@@ -0,0 +1,789 @@
+                                        outstr += '%{' + macroexpr + '}'
+                                lastch = ch
+                                continue
+                        macroexpr += ch
+                    if ch == '{':
+                        if lastch == '%':
+                            if inmacro == 0:
+                                macroexpr = ''
+                                outstr = outstr[:-1]
+                            inmacro += 1
+                    elif ch == '(':
+                        if lastch == '%':
+                            if inshell == 0:
+                                shellexpr = ''
+                                outstr = outstr[:-1]
+                            inshell += 1
+                    if inmacro == 0 and inshell == 0:
+                        if ch == '%':
+                            # Handle unbracketed expressions (in which case we eat the rest of the expression)
+                            if expr[i+1] not in ['{', '%']:
+                                macrokey = expr[i+1:].split()[0]
+                                if macrokey in globaldefs or macrokey in defines or macrokey in values:
+                                    expanded = expand(values.get(macrokey, '') or defines.get(macrokey, '') or globaldefs.get(macrokey, ''))
+                                    if expanded:
+                                        outstr += expanded
+                                        break
+                        if ch == '%' and lastch == '%':
+                            # %% is a literal %, so skip this one (and don't allow this to happen again if the next char is a %)
+                            lastch = ''
+                            continue
+                        outstr += ch
+                    lastch = ch
+                return outstr
+
+            def eval_cond(cond, condtype):
+                negate = False
+                if condtype == '%if':
+                    if cond.startswith('(') and cond.endswith(')'):
+                        cond = cond[1:-1].strip()
+                    if cond.startswith('!'):
+                        cond = cond[1:].lstrip()
+                        negate = True
+                    res = False
+                    try:
+                        if int(cond):
+                            res = True
+                    except ValueError:
+                        pass
+                elif condtype in ['%ifos', '%ifnos']:
+                    if condtype == '%ifnos':
+                        negate = True
+                    # Assume linux
+                    res = ('linux' in cond.split())
+                elif condtype in ['%ifarch', '%ifnarch']:
+                    if condtype == '%ifnarch':
+                        negate = True
+                    res = ('x86_64' in cond.split())
+                else:
+                    raise Exception('Unhandled conditional type "%s"' % condtype)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.